core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/collection/WithinExpression.kt

```diff
@@ -48,25 +50,32 @@ class NotWithinExpression<T : ValidType>(
     override fun toDopeQuery(manager: DopeQueryManager) = toInfixDopeQuery(manager = manager)
 }
 
-fun <T : ValidType> TypeExpression<T>.notWithinArray(collection: TypeExpression<ArrayType<T>>) = NotWithinExpression(this, collection)
-
-fun Number.notWithinArray(collection: TypeExpression<ArrayType<NumberType>>): NotWithinExpression<NumberType> =
-    toDopeType().notWithinArray(collection)
+fun <T : ValidType> TypeExpression<T>.notWithinArray(collection: TypeExpression<ArrayType<T>>) =
+    NotWithinExpression(this, collection)
 
-fun String.notWithinArray(collection: TypeExpression<ArrayType<StringType>>): NotWithinExpression<StringType> =
-    toDopeType().notWithinArray(collection)
+fun Number.notWithinArray(collection: TypeExpression<ArrayType<NumberType>>) = toDopeType().notWithinArray(collection)
 
-fun Boolean.notWithinArray(collection: TypeExpression<ArrayType<BooleanType>>): NotWithinExpression<BooleanType> =
-    toDopeType().notWithinArray(collection)
+fun String.notWithinArray(collection: TypeExpression<ArrayType<StringType>>) = toDopeType().notWithinArray(collection)
 
-fun <T : ValidType> TypeExpression<T>.notWithinArray(collection: Collection<TypeExpression<T>>): NotWithinExpression<T> =
-    this.notWithinArray(collection.toDopeType())
+fun Boolean.notWithinArray(collection: TypeExpression<ArrayType<BooleanType>>) = toDopeType().notWithinArray(collection)
 
-fun Number.notWithinArray(collection: Collection<TypeExpression<NumberType>>): NotWithinExpression<NumberType> =
+fun <T : ValidType> TypeExpression<T>.notWithinArray(collection: Collection<TypeExpression<T>>) =
     notWithinArray(collection.toDopeType())
 
-fun String.notWithinArray(collection: Collection<TypeExpression<StringType>>): NotWithinExpression<StringType> =
-    notWithinArray(collection.toDopeType())
+fun Number.notWithinArray(collection: Collection<TypeExpression<NumberType>>) = notWithinArray(collection.toDopeType())
 
-fun Boolean.notWithinArray(collection: Collection<TypeExpression<BooleanType>>): NotWithinExpression<BooleanType> =
-    notWithinArray(collection.toDopeType())
+fun String.notWithinArray(collection: Collection<TypeExpression<StringType>>) = notWithinArray(collection.toDopeType())
+
+fun Boolean.notWithinArray(collection: Collection<TypeExpression<BooleanType>>) = notWithinArray(collection.toDopeType())
+
+fun <T : ValidType> TypeExpression<T>.notWithinArray(collection: ISelectOffsetClause<T>) =
+    notWithinArray(collection.asExpression())
+
+fun Number.notWithinArray(collection: ISelectOffsetClause<NumberType>) =
+    toDopeType().notWithinArray(collection.asExpression())
```

Comment:
here again (and I guess all over in the project), please rename the param _collection_

Reply:
done